.github/workflows/on_release.yml

```diff
@@ -163,3 +169,94 @@ jobs:
           tags: |
             permitio/opal-client-cedar:latest
             permitio/opal-client-cedar:${{ env.opal_version_tag }}
+
+      - name: Python setup
+        if: ${{ !(github.event_name == 'workflow_dispatch' && github.event.inputs.dry_run == 'true') }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11.8'
+
+      # This is the root file representing the package for all the sub-packages.
+      - name: Bump version - packaging__.py
+        if: ${{ !(github.event_name == 'workflow_dispatch' && github.event.inputs.dry_run == 'true') }}
+        run: |
+          version_tag=${{ github.event.release.tag_name }}
+          version_tuple=$(echo $version_tag | sed 's/\./, /g')
+          sed -i "s/VERSION = (.*/VERSION = (${version_tuple})/" packages/__packaging__.py
+          cat packages/__packaging__.py
+          git config --local user.email "eli@permit.io"
+          git config --local user.name "elimoshkovich"
+          git add packages/__packaging__.py
+          git commit -m "Bump version to ${version_tag}"
```

Comment:
Hmm now I'm thinking - what about pushing it?
Which is also a bit problematic if we're running it after the "release" was created in GitHub (with a tag on a specific commit). 
Not sure what's the right approach here...

Reply:
this is only will push the version update, on the last step of the CI ( in case everything else is succeeded )

Reply:
Oh sorry - I've missed that.
But my other point is still unclear to me - when we'll do the "release", let's say of version `0.8.1`, a new git tag named `0.8.1` would be created.
Only then would this workflow execute, the version in `__packaging__.py` would be bumped and pushed (to `master`?) - although the git tag point to a previous commit that doesn't include the version bump.

Reply:
Maybe a smart solution would be assuming the version is already "bumped" to the right one,
Then bumping the version again only at the end of the workflow (=after already released to pypi) - which would be done in preparation for the next future release (Like on your birthday, one for next year).

As an example:
1. Latest released version is `0.8.1`
2. `__packaging__.py` already points to `0.8.2` (which makes sense because `0.8.2` is now being developed).
3. At some point we decide to release `0.8.2` - a GitHub "Release" is being created and a matching tag is created.
4. Your CI workflow runs, it publishes version `0.8.2` to pypi.
5. At the end of the workflow - the version is bumped to `0.8.3` and pushed to `master`.

How does that sound?

Reply:
Sorry, but I'm not agree with that, there should be one source of truth which is master/main - the __packaging__.py should be exactly the same as the release version.
Some version are in developing process? so it's should be under some branch with the new version you works on.

Reply:
Alright, that's just a suggestion.
But do you understand what I'm trying to solve.
Currently:
**1. You release a version, with title `0.8.2` and matching git tag `0.8.2`. The tag is on a commit where `__packaging__.py` has the older version `0.8.1`**
2. The CI workflow runs and commits a version bump to `__packaging__.py`.
3. You push to pypi with the correct version `0.8.2`.
4. You git push the version bump to `master`. 

You see the issue?